### ISSUE TITLE
Super Mario 63 - Inside the Castle: Main Hall not working

### ISSUE THREAD

**Issue by 68r** [ruffle-rs/ruffle#8217]
### Describe the bug

#What it looks like when you go to the 'inside the castle: main hall" level in super mario 63
![ezgif-2-e7a923b744](https://user-images.githubusercontent.com/90115544/194363984-26daf218-ccf9-483c-aabc-00369e8a76fd.gif)

Debug Info:
# Player Info
Allows script access: true
Renderer: WebGL

# Page Info
Page URL: https://mathgames66.github.io/p/supermario63.html
SWF URL: https://mathgames66.github.io/498969_Super_Mario_63__2009_.swf

# Browser Info
User Agent: Mozilla/5.0 (X11; CrOS x86_64 14909.132.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36
Platform: Linux x86_64
Has touch support: true

# Ruffle Info
Version: 0.1.0
Name: nightly 2022-10-06
Channel: nightly
Built: 2022-10-06T00:17:35.767Z
Commit: 4b2907ead56e5806f441fcb7e32e1030166cb2d2
Is extension: false

# Metadata
width: 450
height: 300
frameRate: 32
numFrames: 25
swfVersion: 8
backgroundColor: #000000
isActionScript3: false
uncompressedLength: 19339460

### Expected behavior

what it supposed to look like: https://super-mario-63.fandom.com/wiki/Main_Hall

### Affected platform

Self-hosted version

### Operating system

Chrome OS 14909 14909.132.0

### Browser

Chrome

**Comment by n0samu** [comment]
Looks like it's hitting a script timeout:
```
[ERROR ruffle_core::avm1::runtime] A script in this movie has taken too long to execute and has been terminated.
[ERROR ruffle_core::avm1::runtime] No more actions will be executed in this movie.
```
I'm not sure if this is a regression from #6906, if it's a difference between game versions, or if Newgrounds has the script timeout disabled. Either way, the performance needs significant improvement.

**Comment by n0samu** [comment]
Fixed by #10330
